includes/api/class-wc-admin-rest-onboarding-profile-controller.php

```diff
@@ -0,0 +1,300 @@
+		}
+
+		$item = $this->prepare_item_for_response( $items, $request );
+		$data = $this->prepare_response_for_collection( $item );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Update onboarding profile data.
+	 *
+	 * @param  WP_REST_Request $request Request data.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function update_items( $request ) {
+		$query_args      = $this->prepare_objects_query( $request );
+		$onboarding_data = get_option( 'wc_onboarding_profile', array() );
+		$update          = update_option( 'wc_onboarding_profile', array_merge( $onboarding_data, $query_args ) );
+
+		if ( $update ) {
+			$result = array(
+				'status'  => 'success',
+				'message' => __( 'Onboarding profile data has been updated.', 'woocommerce-admin' ),
+			);
+		} else {
+			$result = array(
+				'status'  => 'error',
+				'message' => __( 'There was an error updating the onboarding profile data.', 'woocommerce-admin' ),
+			);
+		}
+
+		$response = $this->prepare_item_for_response( $result, $request );
+		$data     = $this->prepare_response_for_collection( $response );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepare objects query.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return array
+	 */
+	protected function prepare_objects_query( $request ) {
+		$args       = array();
+		$properties = self::get_profile_properties();
+
+		foreach ( $properties as $key => $property ) {
+			if ( isset( $request[ $key ] ) ) {
+				$args[ $key ] = $request[ $key ];
+			}
+		}
+
+		/**
+		 * Filter the query arguments for a request.
+		 *
+		 * Enables adding extra arguments or setting defaults for a post
+		 * collection request.
+		 *
+		 * @param array           $args    Key value array of query var to query value.
+		 * @param WP_REST_Request $request The request used.
+		 */
+		$args = apply_filters( 'woocommerce_rest_onboarding_profile_object_query', $args, $request );
+
+		return $args;
+	}
+
+
+	/**
+	 * Prepare the data object for response.
+	 *
+	 * @param object          $item Data object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response $response Response data.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$data     = $this->add_additional_fields_to_object( $item, $request );
+		$data     = $this->filter_response_by_context( $data, 'view' );
+		$response = rest_ensure_response( $data );
+
+		/**
+		 * Filter the list returned from the API.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param array            $item     The original item.
+		 * @param WP_REST_Request  $request  Request used to generate the response.
+		 */
+		return apply_filters( 'woocommerce_rest_prepare_onboarding_profile', $response, $item, $request );
+	}
+
+	/**
+	 * Get onboarding profile properties.
+	 *
+	 * @return array
+	 */
+	public static function get_profile_properties() {
+		$properties = array(
```

Comment:
I noticed none of these had any validation in place (enum) - should we limit some of the values that can be set or is it not important?

Reply:
I was mostly avoiding setting validation since I didn't know if all these values had been finalized yet.  Maybe @justinshreve can chime in here as I think he may know whether they are or not.

Reply:
To the best of my knowledge, the screens there are finalized so this should be our final list of questions, so some validation might not hurt. These questions have also persisted through design updates, it's just the ordering that has changed.

I'll post some notes below:

**Industry**
_Multiselect_
Options:
* Fashion, apparel & accessories
* Health & beauty
* Art, music, & photography
* Food & Drink
* Home, furniture & garden
* Other

**Product Types**
_Multiselect_
Options:
* Physical products
* Downloads
* Subscriptions
* Memberships
* Composite Products
* Spaces
* Rentals

**Product Count**
_Single Selection_
Options:
* 1-10
* 11-100
* 101-1000
* 1000+

**Selling Venues/Already Selling**
_Single Selection_
Options:
* No
* Yes, on another platform
* Yes, at a brick and motor store
* Yes, on another platform and at a brick and mortar store

**Other Platform**
_Single Selection_
Options:
* Shopify
* BigCommerce
* Magento
* Wix
* Other

**Theme**
_Single Selection_
This should probably just be a theme slug, so maybe some simple validation against the slug pattern?